[ADD] electronic_store: Electronic Store

dhrs-odoo opened the pull request:
This module provides essential features to manage an electronic store.
It includes product categories, products, sales orders, and more for electronic products.

Task - 3460473

prra-odoo commented on electronic_store/data/project_project.xml:
```diff
@@ -0,0 +1,55 @@
+    <record id="project_project_4" model="project.project">
+        <field name="name">Refrigerator Installation</field>
+        <field name="allow_billable" eval="True" />
+        <field name="use_documents" eval="True" />
```
```suggestion
```
Compute Field

prra-odoo commented on electronic_store/demo/crm_lead.xml:
```diff
@@ -0,0 +1,85 @@
+        <field name="city">Chicago</field>
+        <field name="country_id" ref="base.us" />
+        <field name="website">https://www.slowfoodchicago.org</field>
+        <field name="date_open" eval="datetime.now()" />
+        <field name="date_closed" eval="datetime.now()" />
```
`date_open` and `date_closed`  shouldn't be the same!

prra-odoo commented on electronic_store/demo/crm_lead.xml:
```diff
@@ -0,0 +1,85 @@
+        <field name="date_open" eval="datetime.now()" />
+        <field name="medium_id" ref="utm.utm_medium_website" />
+        <field name="iap_enrich_done" eval="True" />
+        <field name="description">&lt;p&gt;Can you share the price&lt;/p&gt;</field>
```
CDATA...

prra-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,141 @@
+<odoo>
+    <record id="helpdesk_ticket_1" model="helpdesk.ticket">
+        <field name="name">AC Repair</field>
+        <field name="x_warranty_date">2026-08-17 00:00:00</field>
```
Date fields should have dynamic values

prra-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,141 @@
+        <field name="oldest_unanswered_customer_message_date">2023-09-13 09:06:58</field>
+        <field name="tracking">serial</field>
+    </record>
+</odoo>
```
```suggestion
</odoo>

```

prra-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,141 @@
+        <field name="x_warranty_date">2026-08-17 00:00:00</field>
+        <field name="partner_name">Ruffino's Restaurant</field>
+        <field name="date_last_stage_update">2023-08-16 10:27:30</field>
+        <field name="use_product_returns" eval="True"/>
```
```suggestion
```
Related field

prra-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,141 @@
+        <field name="use_sla" eval="True"/>
+        <field name="assign_date">2023-08-16 10:27:52</field>
+        <field name="open_hours">3478</field>
+        <field name="message_is_follower" eval="True"/>
```
```suggestion
```
Compute Field

prra-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,141 @@
+        <field name="legend_done">Ready</field>
+        <field name="team_id" ref="helpdesk.helpdesk_team1"/>
+        <field name="access_url">/my/ticket/1</field>
+        <field name="has_partner_picking" eval="True"/>
```
```suggestion
```
Compute Field

prra-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,141 @@
+        <field name="team_id" ref="helpdesk.helpdesk_team1"/>
+        <field name="access_url">/my/ticket/1</field>
+        <field name="has_partner_picking" eval="True"/>
+        <field name="legend_normal">In Progress</field>
```
```suggestion
```
Related Field

prra-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,141 @@
+        <field name="access_url">/my/ticket/1</field>
+        <field name="has_partner_picking" eval="True"/>
+        <field name="legend_normal">In Progress</field>
+        <field name="oldest_unanswered_customer_message_date">2023-08-16 10:27:30</field>
```
Dynamic date value

prra-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,141 @@
+        <field name="legend_normal">In Progress</field>
+        <field name="oldest_unanswered_customer_message_date">2023-08-16 10:27:30</field>
+        <field name="tracking">serial</field>
+        <field name="partner_phone">+1 225-753-3458</field>
```
```suggestion
```
Compute Field

prra-odoo reviewed (changes requested):
Hello
Posted my first review, more changes yet to come
Please do the needful changes!

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1724 @@
+                                    class="s_quotes_carousel s_carousel_default carousel slide o_cc o_cc2"
+                                    data-bs-interval="10000" id="myQuoteCarousel715956">
+                                    <ol class="carousel-indicators">
+                                        <li data-bs-slide-to="0" class="active"
+                                            data-bs-target="#myQuoteCarousel715956" />
```
```suggestion
                                        <li data-bs-slide-to="0" class="active" data-bs-target="#myQuoteCarousel715956" />
```
same for other li also

prra-odoo commented on electronic_store/demo/website_view.xml:
Need to improve the formatting in the whole file
remove extra spacings
reduce LOC
FIx indentation
Do proper Formatting

prra-odoo commented on electronic_store/demo/website_page.xml:
```diff
@@ -0,0 +1,27 @@
+        <field name="view_id" ref="services"/>
+        <field name="website_id" ref="website.default_website"/>
+    </record>
+    <record id="website_page_8" model="website.page">
+        <field name="url">/privacy</field>
+        <field name="is_published" eval="True"/>
+        <field name="view_id" ref="website_page_privacy_policy_view"/>
+        <field name="website_id" ref="website.default_website"/>
+    </record>
+    <record id="website_page_9" model="website.page">
+        <field name="url">/contactus</field>
+        <field name="is_published" eval="True"/>
+        <field name="view_id" ref="contactus"/>
+        <field name="website_id" ref="website.default_website"/>
+    </record>
```
Check the reference video and see if unused then please remove both the pages

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                        </div>
+                                    </div>
+                                    <div class="col-lg-2 pt16 pb16 o_colored_level">
+                                        <a href="/contactus" class="btn btn-block btn-primary" data-bs-original-title="" title="">Get in touch</a>
```
```suggestion
                                        <a href="/contactus" class="btn btn-block btn-primary" data-bs-original-title="">Get in touch</a>
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                            data-snippet="s_text_block" style="background-image: none;">
+                            <div class="s_allow_columns container">
+                                <div class="row">
+                                    <div class="col-lg-8 mt-4 mt-lg-0 o_colored_level" style="">
```
```suggestion
                                    <div class="col-lg-8 mt-4 mt-lg-0 o_colored_level">
```
Please make the same change in whole file

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                <t t-call="website.layout">
+                    <t t-set="logged_partner"
+                        t-value="request.env['website.visitor']._get_visitor_from_request().partner_id" />
+                    <t t-set="contactus_form_values"
+                        t-value="{                                 'email_to': res_company.email,                                 'name': request.params.get('name', ''),                                 'phone': request.params.get('phone', ''),                                 'email_from': request.params.get('email_from', ''),                                 'company': request.params.get('company' ...
```
Remove extra spacing

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                                    data-success-mode="redirect"
+                                                    data-success-page="/contactus-thank-you"
+                                                    data-pre-fill="true">
+                                                    <div
+                                                        class="s_website_form_rows row s_col_no_bgcolor">
+                                                        <div
+                                                            class="s_website_form_field mb-3 col-12    s_website_form_required    "
+                                                            data-type="char" data-name="Field">
+                                                            <div
+                                                                class="row s_col_no_resize s_col_no_bgcolor">
```
```suggestion
                                                    <div class="s_website_form_rows row s_col_no_bgcolor">
                                                        <div class="s_website_form_field mb-3 col-12 s_website_form_required" data-type="char" data-name="Field">
                                                            <div class="row s_col_no_resize s_col_no_bgcolor">
```
Please remove extra spacing and fix formatting to reduce LOC

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+        <field name="arch" type="xml">
+            <data name="Contact Form (Opportunity)" inherit_id="website.contactus">
+                <xpath expr="//t[@t-set='contactus_form_values']" position="after">
+                    <t t-set="contactus_form_values"
+                        t-value="dict(contactus_form_values, **{                                         'contact_name': request.params.get('contact_name', ''),                                         'partner_name': request.params.get('partner_name', ''),                                         'description': request.params.get('description', ''),                                 })" />
```
Same over here also

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+        <field name="arch" type="xml">
+            <t name="Helpdesk: Submit a Ticket Form" t-name="website_helpdesk.ticket_submit_form">
+                <div class="container">
+                    <h2 class="text-muted"> Submit a Ticket <t t-if="multiple_teams"> - <t
+                                t-out="team.name" /></t>
```
```suggestion
                    <h2 class="text-muted"> Submit a Ticket <t t-if="multiple_teams"> - <t t-out="team.name" /></t>
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                    <div class="oe_structure" id="oe_structure_whelpdesk_form_1" />
+                    <span class="hidden" data-for="helpdesk_ticket_form"
+                        t-att-data-values="{'team_id': team and team.id or ''}" />
+                    <div id="helpdesk_section" class="">
+
+                    </div>
```
```suggestion
                    <div id="helpdesk_section"></div>
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+        <field name="website_id" ref="website.default_website" />
+        <field name="arch" type="xml">
+            <data>
+                <xpath
+                    expr="//*[hasclass('oe_structure')][@id='oe_structure_website_helpdesk_team_3']"
+                    position="replace">
```
```suggestion
                <xpath expr="//*[hasclass('oe_structure')][@id='oe_structure_website_helpdesk_team_3']" position="replace">
```

prra-odoo reviewed (changes requested):
Hello @assi-odoo 

Some final remarks, afterwards good to go from my side!
Mainly fix formatting and unnecessary spacing and reduce LOC from the `website_views.xml`

Thanks!

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                    </div>
+                                    <div class="col-lg-5 pt24 pb24">
+                                        <h5 class="mb-3">About us</h5>
+                                        <p>We are a team of passionate people whose goal is to improve everyone's life through disruptive products. We build great products to solve your business problems. <br /><br />Our products are designed for small to medium size companies willing to optimize their performance.</p>
```
```suggestion
                                        <p>We are a team of passionate people whose goal is to improve everyone's life through disruptive products. We build great products to solve your business problems.<br/>Our products are designed for small to medium size companies willing to optimize their performance.</p>
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                    </div>
+                                </div>
+                            </section>
+                            <section class="s_title pt40 pb40 o_cc o_cc5" data-vcss="001"
+                                data-snippet="s_title">
```
```suggestion
                            <section class="s_title pt40 pb40 o_cc o_cc5" data-vcss="001" data-snippet="s_title">
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                        <div class="col-lg-6 pt24 pb24">
+                                            <div class="row s_col_no_resize s_col_no_bgcolor">
+                                                <div class="col-lg-4 pb16 o_not_editable" contenteditable="false">
+                                                    <img alt="" src="/web/image/website.s_company_team_image_3"
+                                                        class="img-fluid rounded-circle mx-auto"
+                                                        contenteditable="true" loading="lazy" />
```
```suggestion
                                                    <img alt="" src="/web/image/website.s_company_team_image_3" class="img-fluid rounded-circle mx-auto" contenteditable="true" loading="lazy" />
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                        <div class="col-lg-6 pt24 pb24">
+                                            <div class="row s_col_no_resize s_col_no_bgcolor">
+                                                <div class="col-lg-4 pb16 o_not_editable" contenteditable="false">
+                                                    <img alt="" src="/web/image/website.s_company_team_image_4"
+                                                        class="img-fluid rounded-circle mx-auto"
+                                                        contenteditable="true" loading="lazy" />
```
```suggestion
                                                    <img alt="" src="/web/image/website.s_company_team_image_4" class="img-fluid rounded-circle mx-auto" contenteditable="true" loading="lazy" />
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                class="s_title parallax s_parallax_is_fixed bg-black-50 pt24 pb24"
+                                data-vcss="001" data-snippet="s_title"
+                                data-scroll-background-ratio="1">
+                                <span class="s_parallax_bg oe_img_bg"
+                                    style="background-image: url('/web/image/electronic_store.configurator_1_s_banner_default_image'); background-position: 50% 0;" />
```
```suggestion
                                <span class="s_parallax_bg oe_img_bg" style="background-image: url('/web/image/electronic_store.configurator_1_s_banner_default_image'); background-position: 50% 0;" />
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                <t t-call="website.layout">
+                    <div id="wrap">
+                        <div class="oe_structure">
+                            <section
+                                class="s_title parallax s_parallax_is_fixed bg-black-50 pt24 pb24"
+                                data-vcss="001" data-snippet="s_title"
+                                data-scroll-background-ratio="1">
```
```suggestion
                            <section class="s_title parallax s_parallax_is_fixed bg-black-50 pt24 pb24" data-vcss="001" data-snippet="s_title" data-scroll-background-ratio="1">
```
Same for other sections as well

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                data-vcss="001"
+                                data-oe-shape-data="{'shape':'web_editor/Blocks/04','flip':[]}"
+                                data-snippet="s_three_columns">
+                                <div
+                                    class="o_we_shape o_web_editor_Blocks_04 o_header_extra_shape_mapping" />
```
```suggestion
                                <div class="o_we_shape o_web_editor_Blocks_04 o_header_extra_shape_mapping" />
```
Same goes for other divs as well

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                                <blockquote
+                                                    class="s_blockquote s_blockquote_classic blockquote ms-auto w-50"
+                                                    data-name="Blockquote">
+                                                    <i
+                                                        class="fa fa-1x fa-quote-left rounded float-start bg-o-color-2 s_blockquote_icon" />
```
```suggestion
                                                    <i class="fa fa-1x fa-quote-left rounded float-start bg-o-color-2 s_blockquote_icon" />
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                                                src="/web/image/electronic_store.s_quotes_carousel_demo_image_5"
+                                                                class="s_blockquote_avatar img rounded-circle me-2"
+                                                                alt="" loading="lazy" />
+                                                            <span class="s_blockquote_author"><b>Iris
+                                                                DOE</b> &#8226; CEO of MyCompany</span>
```
```suggestion
                                                            <span class="s_blockquote_author"><b>Iris DOE</b> &#8226; CEO of MyCompany</span>
```

prra-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1669 @@
+                                    <div class="row d-flex align-items-stretch">
+                                        <div class="col-lg-4 s_col_no_bgcolor pt16 pb16">
+                                            <div class="card text-bg-white h-100">
+                                                <img class="card-img-top"
+                                                    src="/web/image/electronic_store.configurator_1_s_three_columns_default_image_1"
+                                                    alt="" loading="lazy" />
```
```suggestion
                                                <img class="card-img-top" src="/web/image/electronic_store.configurator_1_s_three_columns_default_image_1" alt="" loading="lazy" />
```
Same for other Image tags

paso-odoo commented on electronic_store/demo/purchase_order_post.xml:
```diff
@@ -0,0 +1,275 @@
+    
+    <function name="write" model="stock.warehouse">
+        <value eval="[ref('stock.warehouse0')]"/>
+        <value eval="{'name': 'Electronic store', 'partner_id': ref('base.main_partner')}"/>
```
```suggestion
        <value eval="{'name': 'Electronic store'}"/>
```

paso-odoo commented on electronic_store/demo/res_partner.xml:
```diff
@@ -0,0 +1,110 @@
+        <field name="country_id" ref="base.us"/>
+        <field name="zip">70809</field>
+        <field name="website">https://www.ruffinosrestaurant.com</field>
+        <field name="phone">+1 225-753-3458</field>
```
I would use fictive name and email

paso-odoo commented on electronic_store/demo/res_partner.xml:
```diff
@@ -0,0 +1,110 @@
+        <field name="partner_longitude">-93.2912714</field>
+        <field name="website">http://ascs.com</field>
+        <field name="phone">+1 800-234-4725</field>
+        <field name="date_localization" eval="DateTime.today() - relativedelta(days=6)"/>
```
```suggestion
```

paso-odoo commented on electronic_store/demo/sale_order.xml:
```diff
@@ -0,0 +1,75 @@
+<odoo noupdate="1">
+    <record id="sale_order_2" model="sale.order">
+        <field name="partner_id" ref="res_partner_7"/>
+        <field name="date_order" eval="(DateTime.today() - relativedelta(days=7))"/>
+        <field name="warehouse_id" ref="stock.warehouse0"/>
```
```suggestion
```

paso-odoo commented on electronic_store/demo/sale_order.xml:
```diff
@@ -0,0 +1,75 @@
+        <field name="warehouse_id" ref="stock.warehouse0"/>
+        <field name="origin">Slow Food Chicago</field>
+        <field name="opportunity_id" ref="crm_lead_2"/>
+        <field name="commitment_date" eval="(DateTime.today() - relativedelta(years=10))"/>
+        <field name="payment_term_id" ref="account.account_payment_term_immediate"/>
```
```suggestion
```

paso-odoo commented on electronic_store/demo/sale_order_line.xml:
```diff
@@ -0,0 +1,67 @@
+    </record>
+    <record id="sale_order_line_6" model="sale.order.line">
+        <field name="product_id" ref="product_product_1037"/>
+        <field name="qty_to_invoice">2.0</field>
```
```suggestion
```

paso-odoo commented on electronic_store/demo/sale_order_post.xml:
```diff
@@ -0,0 +1,36 @@
+        ]"/>
+    </function>
+
+    <function name="button_validate" model="stock.picking"  context="{'skip_sms': True}">
```
```suggestion
    <function name="button_validate" model="stock.picking" context="{'skip_sms': True}">
```

paso-odoo commented on electronic_store/demo/website.xml:
```diff
@@ -0,0 +1,12 @@
+        <field name="name">kb-electronics</field>
+        <field name="theme_id" ref="base.module_theme_buzzy"/>
+        <field name="configurator_done" eval="True"/>
+        <field name="salesteam_id" ref="sales_team.salesteam_website_sales"/>
+        <field name="crm_default_team_id" ref="sales_team.team_sales_department"/>
```
```suggestion
```
default fields

paso-odoo commented on electronic_store/demo/website_ir_attachment.xml:
```diff
@@ -0,0 +1,73 @@
+        <field name="name">1.jpg</field>
+        <field name="datas" type="base64" file="electronic_store/static/src/binary/ir_attachment/471-1.jpg"/>
+        <field name="key">electronic_store.cover_image</field>
+        <field name="public" eval="True"/>
+        <field name="website_id" ref="website.default_website"/>
```
```suggestion
        <field name="public" eval="True"/>
```

paso-odoo commented on electronic_store/demo/website_theme_apply.xml:
```diff
@@ -0,0 +1,30 @@
+        <value model="ir.module.module" eval="obj().env['ir.module.module'].search([('name', '=', 'theme_buzzy')]).ids"/>
+        <value model="website" eval="obj().env.ref('website.default_website')"/>
+    </function>
+    <function name="write" model="ir.ui.view">
+        <value model="ir.ui.view" eval="obj().env['website'].with_context(website_id=obj().env.ref('website.default_website').id).viewref('website.homepage').id"/>
```
```suggestion
    <function name="write" model="ir.ui.view">
        <value model="ir.ui.view" eval="obj().env['website'].with_context(website_id=ref('website.default_website')).viewref('website.homepage').id"/>
```

paso-odoo commented on electronic_store/demo/website_theme_apply.xml:
```diff
@@ -0,0 +1,30 @@
+        <value model="ir.ui.view" eval="{'arch': obj().env.ref('electronic_store.homepage').arch}"/>
+    </function>
+    <function name="write" model="ir.ui.view">
+        <value model="ir.ui.view" eval="obj().env['website'].with_context(website_id=obj().env.ref('website.default_website').id).viewref('website.contactus').id"/>
```
same as above

paso-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1365 @@
+            </data>
+        </field>
+    </record>
+    <record id="ir_ui_view_2795" model="ir.ui.view">
+        <field name="name">no_autohide_menu</field>
+        <field name="type">qweb</field>
+        <field name="key">website.no_autohide_menu</field>
+        <field name="mode">extension</field>
+        <field name="inherit_id" ref="website.layout"/>
+        <field name="website_id" ref="website.default_website"/>
+        <field name="active" eval="False"/>
+        <field name="arch" type="xml">
+            <data inherit_id="website.layout" active="False">
+                <xpath expr="//header" position="attributes">
+                    <attribute name="t-attf-class" add="#{'o_no_autohide_menu'}" separator=" "/>
+                </xpath>
+            </data>
+        </field>
+    </record>
```
It is useful?

paso-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1365 @@
+        <field name="arch" type="xml">
+            <data inherit_id="website.layout">
+                <xpath expr="//footer//span[hasclass('o_footer_copyright_name')]" position="replace">
+                    <span class="o_footer_copyright_name me-2">Copyright &#169; KB Electronics</span>
```
```suggestion
                    <span class="o_footer_copyright_name me-2">Copyright &#169;Electronics</span>
```

paso-odoo replied:
pending

paso-odoo commented on electronic_store/static/src/binary/ir_attachment/471-1.jpg:
remove duplicate products and attachments

paso-odoo reviewed (changes requested):
@prra-odoo @assi-odoo 

Thanks for working on this!
Here, are some quick reviews for this PR. Please apply them several times if needed!

Cheers!

paso-odoo commented on electronic_store/demo/sale_order.xml:
```diff
@@ -0,0 +1,58 @@
+    </record>
+    <record id="sale_order_3" model="sale.order">
+        <field name="partner_id" ref="res_partner_13"/>
+        <field name="warehouse_id" ref="stock.warehouse0"/>
```
??

paso-odoo commented on electronic_store/demo/sale_order_line.xml:
```diff
@@ -0,0 +1,66 @@
+    </record>
+    <record id="sale_order_line_3" model="sale.order.line">
+        <field name="product_id" ref="product_product_1043"/>
+        <field name="product_uom" ref="uom.product_uom_hour"/>
```
why product_uom?

Please clear all code and remove unnecessary code

paso-odoo commented on electronic_store/demo/sale_order_line.xml:
```diff
@@ -0,0 +1,66 @@
+    </record>
+    <record id="sale_order_line_7" model="sale.order.line">
+        <field name="product_id" ref="product_product_1025"/>
+        <field name="qty_to_invoice">10.0</field>
```
```suggestion
```

paso-odoo commented on electronic_store/demo/website_theme_apply.xml:
```diff
@@ -0,0 +1,30 @@
+        <value eval="{'sequence': 90}"/>
+    </function>
+
+    <record id="website.header_search_box" model="ir.ui.view">
+        <field name="active" eval="False"/>
+    </record>
+    <record id="website.header_text_element" model="ir.ui.view">
+        <field name="active" eval="False"/>
+    </record>
```
add these records in website_view.xml

paso-odoo reviewed (changes requested):
@assi-odoo @dhrs-odoo 

Thanks for working on this PR.
Here are some technical remarks from my side.

Cheers!

dhrs-odoo commented on electronic_store/demo/repair_order.xml:
```diff
@@ -0,0 +1,45 @@
+        <field name="recycle_location_id" ref="stock.stock_location_stock"/>
+        <field name="message_is_follower" eval="True"/>
+        <field name="has_message" eval="True"/>
+        <field name="picking_type_id" ref="repair.picking_type_warehouse0_repair"/>
```
required field

dhrs-odoo commented on electronic_store/data/knowledge_article.xml:
```diff
@@ -0,0 +1,181 @@
+                    <li>Quotation Template</li>
+                    <li>Task Create Based on Sales Order<br/></li>
+                    <li>Payment Terms - Instalment (EMI)</li>
+                    <li>Warranty Tracking On Delivery Date (<font style="color: rgb(156, 0, 255);">Automated Action</font>)</li>
```
I have added one automated action

dhrs-odoo commented on electronic_store/data/knowledge_article.xml:
```diff
@@ -0,0 +1,181 @@
+                            <li>The salesperson sets the "expected delivery date" based on customer availability</li>
+                        </ul>
+                    </li>
+                    <li>The warranty date will be set once the delivery is confirmed (<font style="color: rgb(156, 0, 255);">Studio, Automated Action: Update Warranty Date</font>)<br/></li>
```
I have added one automated action

dhrs-odoo commented:
Hello @paso-odoo 
Given changes has been implemented 
Thank You :innocent:

vava-odoo commented on electronic_store/demo/website_theme_apply.xml:
```diff
@@ -0,0 +1,29 @@
+<odoo>
+    <function name="_theme_load" model="ir.module.module" context="{'apply_new_theme': True}">
+        <value model="ir.module.module" eval="obj().env['ir.module.module'].search([('name', '=', 'theme_buzzy')]).ids"/>
+        <value model="website" eval="obj().env.ref('website.default_website')"/>
```
I see many industries where we use `model="ir.module.module"`, but this seems to make more sense :thinking:

vava-odoo commented on electronic_store/demo/website_theme_apply.xml:
```diff
@@ -0,0 +1,29 @@
+        <value model="website.menu" eval="obj().search([('url', '=', '/helpdesk/customer-care-1'), ('website_id', '=', ref('website.default_website'))]).id"/>
+        <value eval="{'sequence': 90}"/>
+    </function>
+    <function name="unlink" model="website.menu">
+        <value model="website.menu" eval="obj().search([
+            ('website_id', '=', ref('website.default_website')),
+            ('url', '=', '/appointment'),
+        ]).id"/>
+    </function>
```
Shouldn't be necessary, since appointment is not installed
```suggestion
```

vava-odoo commented on electronic_store/demo/x_worksheet_template.xml:
```diff
@@ -0,0 +1,15 @@
+<odoo>
+    <record id="x_project_task_worksheet_template_1_1" model="x_project_task_worksheet_template_1_studio">
+        <field name="x_name">S00002 - Air Conditioner Installation</field>
+        <field name="x_project_task_id" model="project.task" eval="obj().search([('sale_order_id', 'in', [ref('sale_order_2')])]).id"/>
```
I think it's better to use search=""
```suggestion
        <field name="x_project_task_id" model="project.task" search="[('sale_order_id', '=', ref('sale_order_2'))]"/>
```

vava-odoo commented on electronic_store/static/src/binary/res_partner/3-image_1920:
Can't find where it is used

vava-odoo replied:
still

vava-odoo commented on electronic_store/static/src/binary/res_partner/1-image_1920:
Can't find where it is used

vava-odoo replied:
still

vava-odoo commented on electronic_store/data/ir_actions_act_window.xml:
```diff
@@ -0,0 +1,7 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="x_project_task_worksheet_template_1_ir_actions_act_window_1" model="ir.actions.act_window">
```
Where is it used?

dhrs-odoo replied:
It is passed as reference in action_id of worksheet template file

vava-odoo commented on electronic_store/data/base_automation.xml:
```diff
@@ -0,0 +1,10 @@
+        <field name="model_id" ref="stock.model_stock_move_line"/>
+        <field name="filter_domain">["&amp;", ("state", "=", "done"), ("picking_code", "=", "outgoing")]</field>
+        <field name="trigger">on_create_or_write</field>
+        <field name="trigger_field_ids" eval="[(6, 0, [ref('stock.field_stock_move_line__state')])]"/>
```
Does that mean the automated action will be called on all stock move line at any change in one those?

vava-odoo commented on electronic_store/data/product_attribute.xml:
Attributes are useless without variants... I guess this file can be removed

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1355 @@
+                                <div class="o_colored_level col-lg-6">
+                                    <h1 style="text-align: left">&amp; Festival Seasons</h1>
+                                    <p class="lead" style="text-align: left">
+                                        After a year's absence, the free and independent festival is back.
```
This doesn't make any sense in this industry

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1355 @@
+        </t>
+    </template>
+    
+    <record id="ir_ui_view_2542" model="ir.ui.view">
```
I don't see the added value of this view. What did you change?

dhrs-odoo replied:
So the consultant has only made changes in the text so I have made changes accordingly

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1355 @@
+            </data>
+        </field>
+    </record>
+    <record id="ir_ui_view_2555" model="ir.ui.view">
```
Same :thinking:

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1355 @@
+            </t>
+        </field>
+    </record>
+    <record id="website_page_privacy_policy_view" model="ir.ui.view">
```
Looks like the default private policy :thinking:

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1355 @@
+            </t>
+        </field>
+    </record>
+    <record id="ir_ui_view_2788" model="ir.ui.view">
```
Again, not clear what is the added value

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,1355 @@
+            </data>
+        </field>
+    </record>
+    <record id="ir_ui_view_2789" model="ir.ui.view">
```
Same

vava-odoo commented on electronic_store/demo/website_view.xml:
Could have a deeper look at this file. I've the impression that we can remove part of it.

dhrs-odoo commented on electronic_store/data/ir_model_fields.xml:
```diff
@@ -0,0 +1,72 @@
+        <field name="ttype">selection</field>
+        <field name="selection">[('air_conditioner', 'Air Conditioner'), ('refrigerator', 'Refrigerator')]</field>
+    </record>
+    <record id="field_makemodel" model="ir.model.fields">
+        <field name="name">x_makemodel</field>
+        <field name="field_description">Make/Model</field>
+        <field name="model_id" ref="x_project_task_worksheet_template_1_ir_model_1"/>
+        <field name="ttype">char</field>
+    </record>
```
![image](https://github.com/odoo/industry/assets/120459793/fa61479d-3479-403d-ab5b-7b165346bf23)
It is used in the worksheet

vava-odoo replied:
Yes, but what does it mean?!

dhrs-odoo commented on electronic_store/data/ir_ui_view.xml:
```diff
@@ -0,0 +1,93 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="report_custom_x_project_task_worksheet_template_1" model="ir.ui.view">
```
Yes we need to create this view

dhrs-odoo commented on electronic_store/demo/sale_order_line.xml:
```diff
@@ -0,0 +1,57 @@
+        <field name="product_id" ref="product_product_1053"/>
+        <field name="order_id" ref="sale_order_10"/>
+    </record>
+    <record id="sale_order_line_20" model="sale.order.line">
+        <field name="product_id" ref="product_product_1037"/>
+        <field name="order_id" ref="sale_order_11"/>
+    </record>
+    <record id="sale_order_line_21" model="sale.order.line">
+        <field name="product_id" ref="product_product_1043"/>
+        <field name="order_id" ref="sale_order_11"/>
+    </record>
```
We need this sale order line record as products are not set in the sale order

dhrs-odoo commented on electronic_store/demo/sale_order_line.xml:
```diff
@@ -0,0 +1,57 @@
+        <field name="product_id" ref="product_product_1042"/>
+        <field name="order_id" ref="sale_order_4"/>
+    </record>
+    <record id="sale_order_line_11" model="sale.order.line">
+        <field name="product_id" ref="product_product_1023"/>
+        <field name="order_id" ref="sale_order_5"/>
+    </record>
+    <record id="sale_order_line_12" model="sale.order.line">
+        <field name="product_id" ref="product_product_1042"/>
+        <field name="order_id" ref="sale_order_5"/>
+    </record>
```
We need to keep this record as they are not getting set in sale order line if we remove this records

dhrs-odoo commented:
Hello @vava-odoo 
I have implemented the suggested changes,
Kindly have a look at it once you are free
Thank You :innocent:

vava-odoo reviewed (changes requested):
Hi @dhrs-odoo 
Thanks for the update. I listed some other comments, before final validation. 
Cheers!

vava-odoo commented on electronic_store/data/account_analytic_account.xml:
Not done

vava-odoo commented on electronic_store/data/account_analytic_plan.xml:
Not done

dhrs-odoo commented:
Hello @vava-odoo 
I have updated the changes as per your suggestions
Kindly have a look at it 
Thank You and Have a great day ahead :innocent:

vava-odoo commented on electronic_store/data/knowledge_article.xml:
```diff
@@ -0,0 +1,180 @@
+                    <strong style="font-weight: 500;">Flow 4: Field Service Installation - AC</strong>
+                </h2>
+                <p>
+                    In the office, the service manager assign task to installers. To do so
```
```suggestion
                    In the office, the service manager assign tasks to installers. To do so
```

vava-odoo commented on electronic_store/data/knowledge_article.xml:
```diff
@@ -0,0 +1,180 @@
+                    <li>Go to mobile odoo app "Air Conditioner Installation"</li>
+                    <li>Check with My Task</li>
+                    <li>Plan the visit customer location</li>
+                    <li>After completion work update the remarks</li>
```
```suggestion
                    <li>After completion of the work, update the remarks</li>
```

vava-odoo commented on electronic_store/data/knowledge_article.xml:
```diff
@@ -0,0 +1,180 @@
+                </ul>
+                <h2>Flow 5: Website Complaint</h2>
+                <p>
+                    Customer can register their complaint for the existing purchased products with serial numbers from the website. (Follow the Steps) --
```
```suggestion
                    Customer can register their complaint for the existing purchased products with serial numbers from the website:
```

vava-odoo reviewed (changes requested):
Hi @dhrs-odoo 
Thanks for the update. Could you have a look at the following comments?
Cheers!

dhrs-odoo commented on electronic_store/data/project_project.xml:
```diff
@@ -0,0 +1,27 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="project_project_3" model="project.project">
```
Yaa I have updated the stage_id by moving some project in in progress and done state.
We don't need to create a separate file for the project stage.

dhrs-odoo replied:
We don't need it as we are getting task when we confirm the sale order

dhrs-odoo commented:
Hello @vava-odoo 
I have incorporated your comments 
kindly have a look at it once you are free
Thank You and Have a great day ahead :innocent:

dhrs-odoo commented:
Hello @vava-odoo 
I have incorporated the suggested changes 
kindly have a look at it once you are free
Thank You :innocent:

vava-odoo commented on electronic_store/data/knowledge_article.xml:
```diff
@@ -0,0 +1,181 @@
+                    <li>Quotation Template</li>
+                    <li>Task Create Based on Sales Order<br/></li>
+                    <li>Payment Terms - Instalment (EMI)</li>
+                    <li>Warranty Tracking On Delivery Date (<font style="color: rgb(156, 0, 255);">Automated Action</font>)</li>
```
This is still there. No more warranty date tracking, no more automated action, ...

dhrs-odoo replied:
It is an outdated comment as I am not able to see that thing
![image](https://github.com/odoo/industry/assets/120459793/9c2784bf-6efa-488b-adfe-b1e6aaf8eb4b)

vava-odoo replied:
Maybe, but I still see things that should not be here... 
![image](https://github.com/odoo/industry/assets/112173839/a6b6d487-e371-4598-9eef-644fe7111b7f)

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,449 @@
+        <field name="default_code">REF-001</field>
+        <field name="detailed_type">product</field>
+        <field name="expiration_time">1095</field>
+        <field name="optional_product_ids" model="product.template" eval="[(6, 0, [obj().env.ref('electronic_store.product_product_1048').product_tmpl_id.id])]"/>
```
Isn't this working?
```suggestion
        <field name="optional_product_ids" model="product.template" eval="[(6, 0, [ref="product_product_1048_product_template"]"/>
```

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,449 @@
+        <field name="expiration_time">1095</field>
+        <field name="optional_product_ids" model="product.template" eval="[(6, 0, [obj().env.ref('electronic_store.product_product_1048').product_tmpl_id.id])]"/>
+        <field name="use_expiration_date" eval="True"/>
+        <field name="is_published" eval="True"/>
```
:eyes: maybe rebase needed?

vava-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,47 @@
+        <field name="stage_id" ref="helpdesk.stage_new"/>
+        <field name="assign_date"  eval="datetime.today() - relativedelta(months=3)"/>
+        <field name="team_id" ref="helpdesk.helpdesk_team1"/>
+        <field name="oldest_unanswered_customer_message_date" eval="datetime.today() - relativedelta(years=8)"/>
```
No need for that field
```suggestion
```

vava-odoo commented on electronic_store/demo/pos_order_line.xml:
```diff
@@ -0,0 +1,54 @@
+        <field name="product_uom_id" ref="uom.product_uom_unit"/>
+        <field name="margin">100</field>
+        <field name="price_unit">3600</field>
+        <field name="tax_ids_after_fiscal_position" eval="[(6, 0, [ref('account.1_sale_tax_template')])]"/>
```
```suggestion
```

vava-odoo commented on electronic_store/demo/pos_order_line.xml:
```diff
@@ -0,0 +1,54 @@
+        <field name="price_unit">3600</field>
+        <field name="tax_ids_after_fiscal_position" eval="[(6, 0, [ref('account.1_sale_tax_template')])]"/>
+        <field name="price_subtotal_incl">4140</field>
+        <field name="tax_ids" eval="[(6, 0, [ref('account.1_sale_tax_template')])]"/>
```
```suggestion
```

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,886 @@
+        <field name="website_id" ref="website.default_website"/>
+        <field name="active" eval="False"/>
+        <field name="arch" type="xml">
+            <data inherit_id="website.layout" name="Template Header Sidebar" active="False">
```
active false :eyes:

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,886 @@
+        <field name="website_id" ref="website.default_website"/>
+        <field name="active" eval="False"/>
+        <field name="arch" type="xml">
+            <data inherit_id="website.layout" name="Links" active="False">
```
active false :eyes:

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,886 @@
+        <field name="type">qweb</field>
+        <field name="key">website.template_header_sidebar_oe_structure_header_sidebar_1</field>
+        <field name="mode">extension</field>
+        <field name="inherit_id" ref="ir_ui_view_2791"/>
```
Maybe group then

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,886 @@
+        <field name="type">qweb</field>
+        <field name="key">website_sale.template_header_sidebar</field>
+        <field name="mode">extension</field>
+        <field name="inherit_id" ref="ir_ui_view_2791"/>
```
Same

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,886 @@
+        <field name="website_id" ref="website.default_website"/>
+        <field name="active" eval="False"/>
+        <field name="arch" type="xml">
+            <data inherit_id="website.template_header_sidebar" active="False" priority="32">
```
active false

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,886 @@
+        <field name="type">qweb</field>
+        <field name="key">website_helpdesk.team_form_1_oe_structure_whelpdesk_form_1</field>
+        <field name="mode">extension</field>
+        <field name="inherit_id" ref="ir_ui_view_3302"/>
```
same

vava-odoo commented on electronic_store/demo/website_view.xml:
```diff
@@ -0,0 +1,886 @@
+            </data>
+        </field>
+    </record>
+    <record id="website.header_search_box" model="ir.ui.view">
+        <field name="active" eval="False"/>
+    </record>
+    <record id="website.header_text_element" model="ir.ui.view">
+        <field name="active" eval="False"/>
+    </record>
```
:eyes:

vava-odoo reviewed (commented):
Thanks @dhrs-odoo for the update.
Mostly cosmetic changes, except the website views that need to be improved. Maybe check what is standard in 17.0 and can be removed from the module.
Cheers

dhrs-odoo commented:
Hello @vava-odoo 
Given Changes has been implemented 
Thank You :innocent:

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,426 @@
+        <field name="default_code">REF-001</field>
+        <field name="detailed_type">product</field>
+        <field name="expiration_time">1095</field>
+         <field name="optional_product_ids" model="product.template" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```
```suggestion
        <field name="optional_product_ids" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```
+ apply to all

vava-odoo replied:
up

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,426 @@
+        <field name="default_code">REF-002</field>
+        <field name="expiration_time">1095</field>
+        <field name="use_expiration_date" eval="True"/>
+        <field name="optional_product_ids" model="product.template" eval="[(6, 0, [obj().env.ref('electronic_store.product_product_1049').product_tmpl_id.id])]"/>
```
use ref

vava-odoo commented on electronic_store/__manifest__.py:
```diff
@@ -0,0 +1,89 @@
+        'industry_fsm_stock',
+        'knowledge',
+        'point_of_sale',
+        'product_expiry',
```
Product_expiry... Shouldn't we do the same as others with the warranty date, instead?

vava-odoo replied:
up

vava-odoo commented on electronic_store/demo/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,46 @@
+        <field name="stage_id" ref="helpdesk.stage_new"/>
+        <field name="ticket_type_id" ref="helpdesk.type_incident"/>
+        <field name="team_id" ref="helpdesk.helpdesk_team1"/>
+        <field name="oldest_unanswered_customer_message_date" eval="datetime.today() - relativedelta(years=4)"/>
```
please apply the comments to all, this occurence should be removed as well...

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -39,8 +38,7 @@
         <field name="use_expiration_date" eval="True"/>
-        <field name="optional_product_ids" model="product.template" eval="[(6, 0, [obj().env.ref('electronic_store.product_product_1049').product_tmpl_id.id])]"/>
-        <field name="is_published" eval="True"/>
+        <field name="optional_product_ids" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```
```suggestion
        <field name="optional_product_ids" eval="[(6, 0, [ref('product_product_1049_product_template')])]"/>
```

vava-odoo replied:
up

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,426 @@
+        <field name="weight">11.0</field>
+        <field name="standard_price">700.0</field>
+        <field name="expiration_time">1095</field>
+         <field name="optional_product_ids" model="product.template" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```
```suggestion
        <field name="optional_product_ids" eval="[(6, 0, [ref('product_product_1049_product_template')])]"/>
```

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,426 @@
+        <field name="weight">16.0</field>
+        <field name="standard_price">220.0</field>
+        <field name="expiration_time">1095</field>
+         <field name="optional_product_ids" model="product.template" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```
```suggestion
        <field name="optional_product_ids" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,426 @@
+        <field name="default_code">RW-Q18WWZA</field>
+        <field name="expiration_time">1095</field>
+        <field name="use_expiration_date" eval="True"/>
+         <field name="optional_product_ids" model="product.template" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```
```suggestion
        <field name="optional_product_ids" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,426 @@
+        <field name="name">BOSS E111 Portable 125 Watt Hand Blender (Grey)</field>
+        <field name="detailed_type">product</field>
+        <field name="image_1920" type="base64" file="electronic_store/static/src/binary/product_template/89-image_1920.jpg"/>
+         <field name="optional_product_ids" model="product.template" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```
```suggestion
        <field name="optional_product_ids" eval="[(6, 0, [ref('product_product_1048_product_template')])]"/>
```

vava-odoo commented on electronic_store/data/product_product.xml:
```diff
@@ -0,0 +1,426 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="product_product_1048" model="product.product">
+        <field name="name">1 Year Extended Warranty</field>
+        <field name="detailed_type">service</field>
+        <field name="list_price">100.0</field>
+        <field name="image_1920" type="base64" file="electronic_store/static/src/binary/product_template/1049-image_1920"/>
+    </record>
```
There is a weird offset between the id (here 1048) and the image name (here 1049) throughout the whole file. Could you fix that?

dhrs-odoo commented:
Hello @vava-odoo 
Incorporated the suggested changes
Thank You :innocent:

vava-odoo commented on electronic_store/demo/stock_lot.xml:
```diff
@@ -0,0 +1,173 @@
+    <record id="stock_lot_1" model="stock.lot">
+        <field name="name">804KCPY43321</field>
+        <field name="product_id" ref="product_product_1037"/>
+        <field name="location_id" ref="stock.stock_location_customers"/>
```
Maybe remove location_id for all
```suggestion
```

vava-odoo replied:
Are locations useful for others stock lots? :thinking: You can remove all in my opinion

vava-odoo commented on electronic_store/data/ir_model_fields.xml:
```diff
@@ -0,0 +1,64 @@
+        <field name="ttype">char</field>
+    </record>
+    <record id="new_field_to_set_warranty_date" model="ir.model.fields">
+        <field name="name">x_warranty_days</field>
```
Looking at the values, it would make way more sense to have months instead of days. Could you adapt the code and the values?
```suggestion
        <field name="name">x_warranty_months</field>
```

dhrs-odoo commented:
Hello @vava-odoo 
I have completed the given changes
Thank You :innocent:

dhrs-odoo commented:
Hello @vava-odoo 
Done with the changes 
Thank You :innocent:

vava-odoo commented on electronic_store/static/src/binary/website/1-favicon:
Empty?

dhrs-odoo commented:
Hello @vava-odoo 
I have added the x_warranty_date field on lot and helpdesk ticket and also added the server action.
Please have a look at it once you are free
Thank You and Have a great day ahead :innocent:

vava-odoo commented on electronic_store/data/ir_actions_server.xml:
```diff
@@ -0,0 +1,14 @@
+        <field name="code"><![CDATA[
+move_lines = env["stock.move.line"].search([("state", "=", "done"), ("picking_code", "=", "outgoing"), ("lot_id.x_warranty_date", "=", False)])
+for line in move_lines:
+    line.lot_id.write({'x_warranty_date': datetime.datetime.today() + datetime.timedelta(line.product_id.x_warranty_months)})
```
```suggestion
    line.lot_id.write({'x_warranty_date': datetime.datetime.today() + dateutil.relativedelta(months=line.product_id.x_warranty_months)})
```

vava-odoo commented on electronic_store/demo/stock_lot.xml:
```diff
@@ -0,0 +1,173 @@
+    <record id="stock_lot_1" model="stock.lot">
+        <field name="name">804KCPY43321</field>
+        <field name="product_id" ref="product_product_1037"/>
+        <field name="x_warranty_date" eval="DateTime.today() + relativedelta(years=2)"/>
```
This will be computed by the server action
```suggestion
```

vava-odoo reviewed (commented):
Thanks for the update. Small remarks, then you can squash your commits and leeeet's go :rocket:

dhrs-odoo commented:
Hello @vava-odoo 
Just completed the suggested changes and squashed the commits
Thank You :innocent:
